pkg/converter/converter_test.go

```diff
@@ -0,0 +1,160 @@
+		input          string
+		expectedOutput string
+		converterArgs  string
+	}{
+		{
+			name: "simple",
+			input: `apiVersion: install.istio.io/v1alpha1
+kind: IstioOperator
+metadata:
+  name: default
+spec:`,
+			converterArgs: fmt.Sprintf("%s %s -n istio-system -v v1.24.3", istioFile, sailFile),
+			expectedOutput: `apiVersion: sailoperator.io/v1
+kind: Istio
+metadata:
+  name: default
+spec:
+  version: v1.24.3
+  namespace: istio-system`,
+		},
+		{
+			name: "complex",
+			input: `apiVersion: install.istio.io/v1alpha1
+kind: IstioOperator
+metadata:
+  name: default
+spec:
+  components:
+    base:
+      enabled: true
+    pilot:
+      enabled: false
+  values:
+    global:
+      externalIstiod: true
+      operatorManageWebhooks: true
+      configValidation: false
+    base:
+      enableCRDTemplates: true
+    pilot:
+      env:
+        PILOT_ENABLE_STATUS: true`,
+			converterArgs: fmt.Sprintf("%s %s -v v1.24.3 -n istio-system", istioFile, sailFile),
+			expectedOutput: `apiVersion: sailoperator.io/v1
+kind: Istio
+metadata:
+  name: default
+spec:
+  values:
+    global:
+      externalIstiod: true
+      operatorManageWebhooks: true
+      configValidation: false
+    base:
+      enableCRDTemplates: true
+      enabled: true
+    pilot:
+      env:
+        PILOT_ENABLE_STATUS: "true"
+      enabled: false
+  namespace: istio-system
+  version: v1.24.3`,
+		},
+		{
+			name: "mandatory-arguments-only",
+			input: `apiVersion: install.istio.io/v1alpha1
+kind: IstioOperator
+metadata:
+  name: default
+spec:`,
+			converterArgs: istioFile,
+			expectedOutput: `apiVersion: sailoperator.io/v1
+kind: Istio
+metadata:
+  name: default
+spec:
+  namespace: istio-system`,
+		},
+	}
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			g := NewWithT(t)
+			t.Cleanup(func() {
+				g.Expect(os.Remove(istioFile)).To(Succeed())
+				g.Expect(os.Remove(sailFile)).To(Succeed())
+			})
+
+			err := os.WriteFile(istioFile, []byte(tc.input), 0o644)
+			g.Expect(err).To(Succeed(), "failed to write YAML file")
+
+			err = executeConfigConverter(tc.converterArgs)
+			g.Expect(err).To(Succeed(), "error in execution of ./configuration-converter.sh")
+
+			actualOutput, err := os.ReadFile(sailFile)
+			g.Expect(err).To(Succeed(), fmt.Sprintf("Cannot read %s", sailFile))
+
+			actualData, err := parseYaml(actualOutput)
+			g.Expect(err).To(Succeed(), "Failed to parse sailFile")
+
+			expectedData, err := parseYaml([]byte(tc.expectedOutput))
+			g.Expect(err).To(Succeed(), "Failed to parse expected output")
+
+			diff := cmp.Diff(actualData, expectedData)
+			g.Expect(diff).To(Equal(""), "Conversion is not as expected")
+		})
+	}
+}
+
+func executeConfigConverter(scriptArgs string) error {
+	converter := filepath.Join(project.RootDir, "tools", "configuration-converter.sh")
+
+	if _, err := shell.ExecuteCommand(converter + " " + scriptArgs); err != nil {
+		return fmt.Errorf("error in execution of command %w", err)
+	}
+	return nil
```

Comment:
The wrapped error doesn't add any new information, so there's no need to wrap the original error. This can be simplified to:
```suggestion
	_, err := shell.ExecuteCommand(converter + " " + scriptArgs)
	return err
```